compat/mingw.c

```diff
@@ -355,15 +355,19 @@ static inline int needs_hiding(const char *path)
 	if (!*path)
 		return 0;
 
-	for (basename = path; *path; path++)
-		if (is_dir_sep(*path)) {
-			do {
-				path++;
-			} while (is_dir_sep(*path));
-			/* ignore trailing slashes */
-			if (*path)
-				basename = path;
```

Comment:
How about an `else break;` here? That would make your patch substantially simpler.

And yes, this is a bug...

Reply:
Nested loops with single counter are prone to bugs like the one I'm fixing. That's why I decided to also remove the surface where problems arise. If you really want to, I'll change it, but I'd rather submit current patch.

Reply:
If you want to change the shape of the function, that's fine, but please do that as a separate patch that does not claim to fix a bug at the same time. It is hard to spot bugs when a patch tries to do too many things, and I have to admit that I could not convince myself even after reading the patch for a full 2 minutes that it does not introduce a regression...